### ISSUE TITLE
Optimise /firefox/new/ page for people in the manual FF update flow.

### ISSUE THREAD

**Issue by alexgibson** [mozilla/bedrock#11071]
## Description

When a Firefox automatic update checks fail too many times, a prompt appears to manually update Firefox that contains a "Download" button. Clicking on this button will now open a new tab with the following URL: https://www.mozilla.org/en-US/firefox/new/?reason=manual-update.

This new query paremeter will give us much greater insight into why Firefox users are coming to the page to download Firefox again.

When this parameter is present, we should (imho), do our best not to get in the way of what the primary goal clearly is: updating Firefox to the most recent version.

If the `?reason=manual-update` is present we should:

- Not show the "You already have Firefox, now get X" marketing interstitial modal.
- Not enter visitors into experiments.
- It might also be worth firing a custom GA event, so we can measure how many people we're converting?

Related bugs:

- https://bugzilla.mozilla.org/show_bug.cgi?id=1746517
- https://bugzilla.mozilla.org/show_bug.cgi?id=1730110

**Comment by alexgibson** [comment]
Over the past 30 days we've seen over 344,000 people hit this flow, which seems like a lot. Maybe we should consider doing something here?

**Comment by slightlyoffbeat** [comment]
@alexgibson with that large of a number every month, a stripped down, simplified page would be best. I will add it to the UX backlog.

In the meantime, lets move forward with the first 2 considerations you presented (removing interstitial and not entering into experiments).

**Comment by pmac** [comment]
Does this mean that the [installer-help](https://www.mozilla.org/en-US/firefox/installer-help/) page is no longer used? Or is that page specific to only the stub installer?

**Comment by slightlyoffbeat** [comment]
@pmac Alex can confirm, but I that page is for a download issues, and not auto update issues?

Could this page be used with slightly different copy instead of /new ? There's a UX question here as well as it may not be best to ask them to choose between 4 browsers when they really just need to download their main browser again.

**Comment by alexgibson** [comment]
@pmac @slightlyoffbeat that's right, installer-help is for a failed stub installer flow. This param is part of a failed auto-update flow.

> Could this page be used with slightly different copy instead of /new ? There's a UX question here as well as it may not be best to ask them to choose between 4 browsers when they really just need to download their main browser again.

We could show a custom template here yes. These are good UX questions so I'll try and find out more in the originating bug.

**Comment by dveditz** [comment]
I added some information to https://bugzil.la/1746517 on how to test that flow in Firefox

**Comment by dveditz** [comment]
> Over the past 30 days we've seen over 344,000 people hit this flow, which seems like a lot. Maybe we should consider doing something here?

Does that represent a recent spike? Or is that volume fairly consistent since we added the ?reason=manual-update parameter?

Do we have any analytics that suggest whether that's 344,000 unique visitors, or if some of the same people keep hitting the problem?

What versions do they appear to have? Are they relatively up-to-date and having new problems? Or are there a lot of old versions who have clearly been having trouble updating for a while? Any unusual correlations between platforms or regions or versions or anything else that differs from our usual traffic or from the profile of our installed base?

**Comment by alexgibson** [comment]
> Does that represent a recent spike? Or is that volume fairly consistent since we added the ?reason=manual-update parameter?

The traffic definitely comes in spikes, which coincides with Firefox updates most likely. On low days we can see ~10,000 page views, going up to as high as 68,000 per day. Since March 4th, we have not see lower than 9,000 page views on any given day.

> Do we have any analytics that suggest whether that's 344,000 unique visitors, or if some of the same people keep hitting the problem?

It's actually worse then I first noticed, as my initial report was for en-US only. Looking across all locales, we have seen over 1 million page views for `/firefox/new/?reason=manual-update` over the last 30 days. GA says over 933,000 of these were unique page views, from over 719,000 individual users.

> What versions do they appear to have? Are they relatively up-to-date and having new problems? Or are there a lot of old versions who have clearly been having trouble updating for a while? Any unusual correlations between platforms or regions or versions or anything else that differs from our usual traffic or from the profile of our installed base?

It's a little hard to tell across all locales, but just picking a few key locales to drill down into (such as en-US, de, fr) shows that the vast majority are on very recent Firefox versions (for en-US 69% are on 99.0, 17% on 98.0, 10% on 97.0). There only seems to be a short tail on old or outdated versions. The vast majority of effected users are also on Windows (for en-US Windows 7 = 58%, Windows 10 = 35%). I see wide mix of countries effected (India, Germany, Brazil, United States, France being the top 5).

**Comment by alexgibson** [comment]
> Does that represent a recent spike?

Another notable thing: we started seeing this data on February 8th, however it wasn't until March 5th that we saw a notable uptick. Since then, traffic has been consistently higher. And we have seen several large spikes (March 15th, April 13th, May 4th being the biggest).

![image](https://user-images.githubusercontent.com/400117/167637889-794b96f6-47d8-4d17-b224-80892f0e4f76.png)

**Comment by pmac** [comment]
After talking about this with some more people we're thinking it's possible that if people have an issue updating, then we should not be offering them the stub installer so we can avoid the possibility of that process also failing and sending them to yet another error page on the site. Can we force full download when we get the `?reason` param?

**Comment by alexgibson** [comment]
> Can we force full download when we get the ?reason param?

I think so yes, we could certainly do this in the interim. Although, if we're also talking about designing a custom page for this flow, then offering the full installer could be a requirement there.

**Comment by alexgibson** [comment]
@slightlyoffbeat just to follow up here:

> There's a UX question here as well as it may not be best to ask them to choose between 4 browsers when they really just need to download their main browser again.

From info gathered in the originating bug, it seems that pre-release users are sent to different locations (e.g. `/firefox/channel/`) when they hit this issue, so if we design a page here for when someone hits `/firefox/new/?reason=manual-update`, then we can be confident they are on the main release version of Firefox, which should hopefully simplify UX.

**Comment by alexgibson** [comment]
> Can we force full download when we get the ?reason param?

@pmac thinking about this a bit more, offering the full installer could well complicate things here a little, as we typically rely on the stub installer to figure out if people need 32/64bit installers.

Is there still a requirement for full installers here, or was this question semi related to https://github.com/mozilla/bedrock/issues/11606?
